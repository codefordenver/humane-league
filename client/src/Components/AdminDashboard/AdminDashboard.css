.AdminDashboard {
  display: flex;
  flex-direction: column;
  align-items: center; }
  .AdminDashboard .admin-greeting {
    color: #fff;
    font-size: 30pt; }
  .AdminDashboard .main-dashboard {
    display: flex;
    justify-content: space-between;
    width: 75%; }
    .AdminDashboard .main-dashboard .dashboard-section {
      display: flex;
      flex-direction: column;
      width: 400px;
      margin: 10px; }
      .AdminDashboard .main-dashboard .dashboard-section h2 {
        text-align: center;
        font-size: 24pt; }
