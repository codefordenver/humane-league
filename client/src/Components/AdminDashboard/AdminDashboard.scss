.AdminDashboard {
  display: flex;
  flex-direction: column;
  align-items: center;
  .admin-greeting {
    color: #fff;
    font-size: 30pt;
    margin-bottom: 75px;
  }
  .main-dashboard {
    display: flex;
    justify-content: space-between;
    width: 75%;
    .dashboard-section {
      display: flex;
      flex-direction: column;
      width: 400px;
      margin: 10px;
      h2 {
        color: #999;
        font-size: 24pt;
        text-align: center;
      }
      .admin-btn {
        background: #1CB841;
        border: none;
        border-radius: 5px;
        box-shadow: 2px 2px 15px rgba(0, 0, 0, 0.6);
        color: #000;
        font-size: 20pt;
        text-decoration: none;
        display: flex;
        align-items: center;
        justify-content: center;
        height: 70px;
        margin: 10px;
        transition: 0.3s;
        &:hover {
          color: white;
          background: #009639;
          box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.6);
        }
      }
    }
  }
}