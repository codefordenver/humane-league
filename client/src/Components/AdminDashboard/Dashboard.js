import React from 'react';
import { NavLink } from 'react-router-dom';
// import './Dashboard.css';

const Dashboard = () => {
  return (
    <section className='main-dashboard'>
      <h1 className='admin-greeting'>Admin Dashboard</h1>
      
      <div className="dashboard-container">
        <div className='dashboard-section'>
          <h2>MANAGE ACTIONS</h2>

          <NavLink 
            to='/admin/create' 
            className='admin-btn create-new-action'>
            Create New Action</NavLink>
          <NavLink 
            to='/admin/update'
            className='admin-btn enable-disable'>
            Edit/Update Actions</NavLink>
        </div>
        <div className='dashboard-section'>
          <h2>ANALYTICS</h2>

          <NavLink 
            to='/admin/log'
            className='admin-btn action-log'>
            Action Log</NavLink>
          <NavLink 
            to='/admin/feedback'
            className='admin-btn user-feedback'>
            User Feedback</NavLink>
        </div>
      </div>
    </section>
  );
};

export default Dashboard;