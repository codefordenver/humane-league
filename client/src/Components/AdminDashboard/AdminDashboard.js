/* eslint-disable import/no-named-as-default*/

import React, { Component } from 'react';
import { Switch, Route } from 'react-router';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Dashboard from './Dashboard';
import CreateNewAction from '../AdminForms/CreateNewAction';
import UpdateAction from '../AdminForms/UpdateAction';
import AddContent from '../AdminForms/AddContent';
import ActionLog from '../AdminForms/ActionLog';
import UserFeedback from '../AdminForms/UserFeedback';
import AdminActions from '../AdminActions/AdminActions';
import './AdminDashboard.css';

class AdminDashboard extends Component {
  componentDidMount() {
    if (!this.props.User.admin) {
      this.props.history.push('/home');
    }
  }

  render() {
    return (
      <div className='AdminDashboard'>
        <Switch>
          <Route exact path='/admin' component={Dashboard} />
          <Route path='/admin/create' component={CreateNewAction} />
          <Route path='/admin/update' component={UpdateAction} />
          <Route path='/admin/addcontent' component={AddContent} />
          <Route path='/admin/log' component={ActionLog} />
          <Route path='/admin/feedback' component={UserFeedback} />
          <Route path='/admin/actions' component={AdminActions} />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = (store) => ({
  User: store.User 
});

export default connect(mapStateToProps, null)(AdminDashboard);

AdminDashboard.propTypes = {
  User: PropTypes.object,
  history: PropTypes.object
};