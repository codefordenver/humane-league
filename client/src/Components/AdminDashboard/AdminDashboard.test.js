import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';

import AdminDashboard from './AdminDashboard';
import Dashboard from './Dashboard';

window.fetch = jest.fn().mockImplementation( () => Promise.resolve({
  json: () => Promise.resolve({
    status: 'Success'
  })
}));

describe('AdminDashboard component tests', () => {
  let renderedAdminDashboard;

  beforeEach(() => {
    renderedAdminDashboard = shallow(
      <MemoryRouter initialEntries={[ { pathname: '/', key: 'testKey' } ]}>
        <AdminDashboard />
      </MemoryRouter>
    )
  })
  it('renders without crashing', () => {
    expect(renderedAdminDashboard).toBeDefined();
  });
  it('matches SnapShot', () => {
    expect(renderedAdminDashboard).toMatchSnapshot();
  }); 
});

describe('Dashboard component tests', () => {
  let renderedDashboard;
  
  beforeEach(() => {
    renderedDashboard = shallow( <Dashboard /> );
  })
  it('renders without crashing', () => {
    expect(renderedDashboard).toBeDefined();
  });
  it('matches SnapShot', () => {
    expect(renderedDashboard).toMatchSnapshot();
  })
})