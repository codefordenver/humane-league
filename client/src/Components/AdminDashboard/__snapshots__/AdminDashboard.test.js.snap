// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminDashboard component tests matches SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MemoryRouter
    initialEntries={
        Array [
            Object {
              "key": "testKey",
              "pathname": "/",
            },
          ]
    }
>
    <Connect(AdminDashboard) />
</MemoryRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Connect(AdminDashboard) />,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(AdminDashboard) />,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Dashboard component tests matches SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="admin-greeting"
>
          Admin Dashboard
</h1>,
        <div
          className="dashboard-container"
>
          <div
                    className="dashboard-section"
          >
                    <h2>
                              MANAGE ACTIONS
                    </h2>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="admin-btn create-new-action"
                              to="/admin/create"
                    >
                              Create New Action
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="admin-btn enable-disable"
                              to="/admin/update"
                    >
                              Edit/Update Actions
                    </NavLink>
          </div>
          <div
                    className="dashboard-section"
          >
                    <h2>
                              ANALYTICS
                    </h2>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="admin-btn action-log"
                              to="/admin/log"
                    >
                              Action Log
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="admin-btn user-feedback"
                              to="/admin/feedback"
                    >
                              User Feedback
                    </NavLink>
          </div>
</div>,
      ],
      "className": "main-dashboard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Admin Dashboard",
          "className": "admin-greeting",
        },
        "ref": null,
        "rendered": "Admin Dashboard",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="dashboard-section"
>
              <h2>
                            MANAGE ACTIONS
              </h2>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="admin-btn create-new-action"
                            to="/admin/create"
              >
                            Create New Action
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="admin-btn enable-disable"
                            to="/admin/update"
              >
                            Edit/Update Actions
              </NavLink>
</div>,
            <div
              className="dashboard-section"
>
              <h2>
                            ANALYTICS
              </h2>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="admin-btn action-log"
                            to="/admin/log"
              >
                            Action Log
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="admin-btn user-feedback"
                            to="/admin/feedback"
              >
                            User Feedback
              </NavLink>
</div>,
          ],
          "className": "dashboard-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  MANAGE ACTIONS
</h2>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="admin-btn create-new-action"
                  to="/admin/create"
>
                  Create New Action
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="admin-btn enable-disable"
                  to="/admin/update"
>
                  Edit/Update Actions
</NavLink>,
              ],
              "className": "dashboard-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "MANAGE ACTIONS",
                },
                "ref": null,
                "rendered": "MANAGE ACTIONS",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Create New Action",
                  "className": "admin-btn create-new-action",
                  "to": "/admin/create",
                },
                "ref": null,
                "rendered": "Create New Action",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Edit/Update Actions",
                  "className": "admin-btn enable-disable",
                  "to": "/admin/update",
                },
                "ref": null,
                "rendered": "Edit/Update Actions",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  ANALYTICS
</h2>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="admin-btn action-log"
                  to="/admin/log"
>
                  Action Log
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="admin-btn user-feedback"
                  to="/admin/feedback"
>
                  User Feedback
</NavLink>,
              ],
              "className": "dashboard-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ANALYTICS",
                },
                "ref": null,
                "rendered": "ANALYTICS",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Action Log",
                  "className": "admin-btn action-log",
                  "to": "/admin/log",
                },
                "ref": null,
                "rendered": "Action Log",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "User Feedback",
                  "className": "admin-btn user-feedback",
                  "to": "/admin/feedback",
                },
                "ref": null,
                "rendered": "User Feedback",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="admin-greeting"
>
            Admin Dashboard
</h1>,
          <div
            className="dashboard-container"
>
            <div
                        className="dashboard-section"
            >
                        <h2>
                                    MANAGE ACTIONS
                        </h2>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="admin-btn create-new-action"
                                    to="/admin/create"
                        >
                                    Create New Action
                        </NavLink>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="admin-btn enable-disable"
                                    to="/admin/update"
                        >
                                    Edit/Update Actions
                        </NavLink>
            </div>
            <div
                        className="dashboard-section"
            >
                        <h2>
                                    ANALYTICS
                        </h2>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="admin-btn action-log"
                                    to="/admin/log"
                        >
                                    Action Log
                        </NavLink>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="admin-btn user-feedback"
                                    to="/admin/feedback"
                        >
                                    User Feedback
                        </NavLink>
            </div>
</div>,
        ],
        "className": "main-dashboard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Admin Dashboard",
            "className": "admin-greeting",
          },
          "ref": null,
          "rendered": "Admin Dashboard",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="dashboard-section"
>
                <h2>
                                MANAGE ACTIONS
                </h2>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="admin-btn create-new-action"
                                to="/admin/create"
                >
                                Create New Action
                </NavLink>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="admin-btn enable-disable"
                                to="/admin/update"
                >
                                Edit/Update Actions
                </NavLink>
</div>,
              <div
                className="dashboard-section"
>
                <h2>
                                ANALYTICS
                </h2>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="admin-btn action-log"
                                to="/admin/log"
                >
                                Action Log
                </NavLink>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="admin-btn user-feedback"
                                to="/admin/feedback"
                >
                                User Feedback
                </NavLink>
</div>,
            ],
            "className": "dashboard-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    MANAGE ACTIONS
</h2>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="admin-btn create-new-action"
                    to="/admin/create"
>
                    Create New Action
</NavLink>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="admin-btn enable-disable"
                    to="/admin/update"
>
                    Edit/Update Actions
</NavLink>,
                ],
                "className": "dashboard-section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "MANAGE ACTIONS",
                  },
                  "ref": null,
                  "rendered": "MANAGE ACTIONS",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Create New Action",
                    "className": "admin-btn create-new-action",
                    "to": "/admin/create",
                  },
                  "ref": null,
                  "rendered": "Create New Action",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Edit/Update Actions",
                    "className": "admin-btn enable-disable",
                    "to": "/admin/update",
                  },
                  "ref": null,
                  "rendered": "Edit/Update Actions",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    ANALYTICS
</h2>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="admin-btn action-log"
                    to="/admin/log"
>
                    Action Log
</NavLink>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="admin-btn user-feedback"
                    to="/admin/feedback"
>
                    User Feedback
</NavLink>,
                ],
                "className": "dashboard-section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ANALYTICS",
                  },
                  "ref": null,
                  "rendered": "ANALYTICS",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Action Log",
                    "className": "admin-btn action-log",
                    "to": "/admin/log",
                  },
                  "ref": null,
                  "rendered": "Action Log",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "User Feedback",
                    "className": "admin-btn user-feedback",
                    "to": "/admin/feedback",
                  },
                  "ref": null,
                  "rendered": "User Feedback",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
